docu/Concepts/TechnicalRequirements/UC_Introduction_of_Gradido-ID.md

```diff
@@ -0,0 +1,141 @@
+
+The second step is to decribe all concerning business logic processes, which have to be adapted by introducing the Gradido-ID.
+
+### Database-Schema
+
+#### Users-Table
+
+The entity users has to be changed by adding the following columns.
+
+| Column                   | Type   | Description                                                                            |
+| ------------------------ | ------ | -------------------------------------------------------------------------------------- |
+| gradidoID                | String | technical unique key of the user as UUID (version 4)                                   |
+| alias                    | String | a business unique key of the user                                                      |
+| passphraseEncryptionType | int    | defines the type of encrypting the passphrase: 1 = email (default), 2 = gradidoID, ... |
+| emailID                  | int    | technical foreign key to the new entity Contact                                        |
+
+##### Email vs emailID
+
+The existing column `email`, will now be changed to the primary email contact, which will be stored as a contact entry in the new `UserContacts` table. It is necessary to decide if the content of the `email `will be changed to the foreign key `emailID `to the contact entry with the email address or if the email itself will be kept as a denormalized and duplicate value in the `users `table.
+
+The preferred and proper solution will be to add a new column `Users.emailId `as foreign key to the `UsersContact `entry and delete the `Users.email` column after the migration of the email address in the `UsersContact `table.
+
+#### new UserContacts-Table
+
+A new entity `UserContacts `is introduced to store several contacts of different types like email, telephone or other kinds of contact addresses.
+
+| Column          | Type   | Description                                                                                                                                                            |
+| --------------- | ------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| id              | int    | the technical key of a contact entity                                                                                                                                  |
+| type            | int    | Defines the type of contact entry as enum: Email, Phone, etc                                                                                                           |
+| userID          | int    | Defines the foreign key to the `Users` table                                                                                                                         |
```

Comment:
this field was called `GradidoID` before

Reply:
not correct.
the userID is the foreign key from UsersContact tabel to the Users table.
The used pattern for foreignkey naming is
<referred table name>ID

-> so this attribut will be renamed from userID to usersID